int32 L_tick
int32 R_tick